### PR TITLE
test/add-tests

### PR DESCRIPTION
**Pull request by tienshaoku**
### PR Reminders

Be ware of the followings

- [ ] implement deployment script for your changes in deployment repo
- [ ] add verification in hardhat simulation(000-prepare-simulation-check, 902-newMarket-check or 903-simulation-check) and system test(901-system-test) in deployment repo
- [ ] update change log
 
---

1. **Contract**: make sure the code follows our convention, ex: naming, explicit returns, etc.; if uncertain, discuss with others

2. **Test**: make sure tests can cover most normal and edge cases; if not, open follow-up tickets. Also, look out for failed tests and fix them!

3. **CHANGELOG.md**: update when external interfaces are changed

4. Workflow: 
    - Github: assign the pr to yourself; if pairing, can merge directly; else, assign someone to help review
    - Asana: assign the corresponding ticket to yourself and leave the pr link on it for easier follow-ups
    - Discord: if someone is mentioned in this pr, tag on Discord

### CODE REVIEW

File: test/foundry/helper/Helper.sol [hunk 1]
```diff
@@ -0,0 +1,8 @@
+pragma solidity 0.7.6;
+import "forge-std/Test.sol";
+
+contract Helper is Test {
+    function _expect_event_emitted_from(address from) internal {
+        vm.expectEmit(false, false, false, true, address(from));
```
Comment (42bchen) [inline test/foundry/helper/Helper.sol]: in some cases, it will be out of our expectations if only check data instead of the indexed part

File: test/foundry/marketRegistry/MarketRegistry.t.sol [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        // add first pool
+        vm.mockCall(
+            address(pool),
+            abi.encodeWithSelector(IUniswapV3PoolState.slot0.selector),
+            abi.encode(100, 0, 0, 0, 0, 0, false)
+        );
+
+        _expect_event_emitted_from(address(marketRegistry));
+        emit PoolAdded(address(baseToken), _DEFAULT_POOL_FEE, address(pool));
+        marketRegistry.addPool(address(baseToken), _DEFAULT_POOL_FEE);
```
Comment (42bchen) [inline test/foundry/marketRegistry/MarketRegistry.t.sol:27]: if we use `baseToken2` here, the test will pass as well, but we expect it will fail

**Review by 42bchen — CHANGES_REQUESTED** [review]

**Review by 42bchen — APPROVED** [review]
LGTM
